### PR TITLE
firehose support dynamic partitioning

### PR DESCRIPTION
**Pull request by patelpratik4job**
*Issue #, if available:*
Existing library do not support new dynamic partitioning for firehose! 

*Description of changes:*
added metadata in firehose response record type to support dynamic partitioning

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: events/testdata/kinesis-firehose-response.json [hunk 1]
```diff
@@ -3,12 +3,29 @@
...
+       }
+     },
+     {
+       "data": "SGVsbG8gV29ybGQ=",
+       "recordId": "record3",
+       "result": "TransformedStateOk",
+       "metadata": {
+         "partitionKeys": {
+           "iamKey1": "iamValue1",
+           "iamKey2": "iamValue2"
```
Comment (bmoffatt) [inline events/testdata/kinesis-firehose-response.json:26]: Can you help me find the related AWS documentation for this feature? I want to understand if `interface{}` is truly the right type for this field vs `map[string]interface{}` or `map[string]string`

Comment (patelpratik4job) [reply events/testdata/kinesis-firehose-response.json:26]: here is the link to documentation and sample code example written in python -> https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html

And here is more information - blog post on firehose dynamic partitioning -> https://aws.amazon.com/blogs/big-data/kinesis-data-firehose-now-supports-dynamic-partitioning-to-amazon-s3/

Comment (bmoffatt) [reply events/testdata/kinesis-firehose-response.json:26]: Thanks! It appears to me from those examples that it'd be correct to instead define the partition keys type like this

```
type KinesisFirehoseResponseRecordMetadata struct {
	PartitionKeys map[string]string `json:"partitionKeys"`
}
```

Comment (patelpratik4job) [reply events/testdata/kinesis-firehose-response.json:26]: ok - let me make that change and test it out and re submit PR

Comment (patelpratik4job) [reply events/testdata/kinesis-firehose-response.json:26]: @bmoffatt changed PartitionKeys data type to map[string]string - also tested code in sandbox environment - it works

**Review by bmoffatt — APPROVED** [review]

**Pull request merged**